src/v/cluster/controller_backend.cc

```diff
@@ -1919,6 +1926,12 @@ controller_backend::shutdown_partition(ss::lw_shared_ptr<partition> partition) {
           ntp, gr, partition->get_log_revision_id());
         // shutdown partition
         co_return co_await _partition_manager.local().shutdown(ntp);
+    } catch (const seastar::gate_closed_exception&) {
+        vlog(
+          clusterlog.debug,
+          "error shutting down {} partition, app shutting down",
+          ntp);
+        co_return make_error_code(errc::shutting_down);
```

Comment:
Nit: any difference between this and just allowing the exception to bubble up?

Reply:
It will be different: `controller_backend::try_reconcile_ntp` won't remember or print the error but will break the loop, Probably even better, will change.